### PR TITLE
REF: check instead of catching TypeError

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/base.py [hunk 1]
```diff
@@ -602,9 +602,9 @@ def _aggregate_multiple_funcs(self, arg, _level, _axis):
         if not len(results):
             raise ValueError("no results")
 
-        try:
+        if all(np.ndim(x) > 0 for x in results):
             return concat(results, keys=keys, axis=1, sort=False)
-        except TypeError:
```
Comment (jorisvandenbossche) [inline pandas/core/base.py]: What's an example that now ran into this TypeError?

Comment (jbrockmendel) [reply pandas/core/base.py]: > What's an example that now ran into this TypeError?

`results = [3, 2, 1.5]; keys = ['size', 'count', 'mean']`

Comment (jorisvandenbossche) [reply pandas/core/base.py]: Sorry, I meant the original dataframe / groupby example or test case that gives those values at this location

**Review by jreback — APPROVED** [review]

File: pandas/core/base.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/base.py]: Identifying some relevant cases with an `assert False`, the place to look is in tests.frame.test_apply, e.g. test_non_callable_aggregates

**Comment by jreback** [comment]
thanks @jbrockmendel

**Comment by jorisvandenbossche** [comment]
@jreback if I review a PR with comment, can you try to give me a little bit of time to respond to Brock's response?

@jbrockmendel thanks for that reference. So in that test case, you have functions that return single numbers. 
But, you can also have an aggregation that returns an array-like or list, which has a dimension, but cannot be concatted. Quick dummy example with the case from the tests:

```
In [8]: df = DataFrame( 
   ...:     {"A": [None, 2, 3], "B": [1.0, np.nan, 3.0], "C": ["foo", None, "bar"]} 
   ...: )  

In [9]: df.agg([lambda x: list(x.unique())]) 

Out[9]: 
                        A                B                 C
<lambda>  [nan, 2.0, 3.0]  [1.0, nan, 3.0]  [foo, None, bar]
```

This now fails. 
(the above example looks a bit unnatural, but in general, returning lists from an aggregation is something that certainly gets done I think, also in groupby context)

**Comment by jreback** [comment]
@jorisvandenbossche i gave a day here which is reasonable
if you want to block then put a hold on the review

**Comment by jbrockmendel** [comment]
@jreback while i agree an explicit "hold" would be helpful, I think Joris gets some extra slack because of the timezone differences.

@jorisvandenbossche have i mentioned recently how much I dislike listlike corner cases?  But the case you brought is Technically Correct, so I'd be OK if you want to revert (and add that test case)

**Comment by jorisvandenbossche** [comment]
> @jreback while i agree an explicit "hold" would be helpful, I think Joris gets some extra slack because of the timezone differences.

Indeed ;) For me, the "1 day" (15 hours since Brocks comment) was just a morning 
Maybe just interpret a comment that is not accompanied with a github approve as a sign that I want to check back?

> But the case you brought is Technically Correct, so I'd be OK if you want to revert (and add that test case)

Do you directly think of another way to do this that you would prefer more than the try/except? Otherwise I can indeed do a revert yes.

**Comment by jbrockmendel** [comment]
> Do you directly think of another way to do this that you would prefer more than the try/except?

The check in this PR was the best idea I had.  Do you have any suggestions?

**Comment by jorisvandenbossche** [comment]
We could check for actual series/frame objects (since concat only allows that), but personally, I find this a perfect use case for the try except.

**Comment by jorisvandenbossche** [comment]
Added a test case here: https://github.com/pandas-dev/pandas/pull/29632

**Pull request merged**